### PR TITLE
carrot2 4.0.0

### PR DESCRIPTION
**Pull request by ankane**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Notes
- The build expects a git directory, so fails with tarball
- Bottles are needed, so let me know if `bottle :unneeded` needs manually removed

cc #58637 

Edit: Sorry for the noise - realized it was better to use the root directory of the dist instead of the `dcs` subdirectory.

### CODE REVIEW

**Comment by ankane** [comment]
The build shouldn’t take this long, so probably best to cancel it.

**Comment by chenrui333** [comment]
cancelled the build, indeed, it takes a little bit too long.

**Comment by ankane** [comment]
Looks like restoring `exec` fixed it from hanging on the `test` step (doesn't hang locally but does on Homebrew CI).

File: Formula/carrot2.rb [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
   desc "Search results clustering engine"
   homepage "https://project.carrot2.org"
-  url "https://github.com/carrot2/carrot2/releases/download/release%2F3.16.3/carrot2-dcs-3.16.3.zip"
-  sha256 "653221f8d11a5712f6889555110ffb4b8eab9ba1ac042cb35a5a16f4531e5ee1"
+  url "https://github.com/carrot2/carrot2.git",
+      tag:      "release/4.0.0",
+      revision: "6a5e2ff984b3ec60375fd475c7cdcd25f7403beb"
   license "Apache-2.0"
 
   bottle :unneeded
```
Comment (SeekingMeaning) [inline Formula/carrot2.rb:9]:
```suggestion
```

**Comment by ankane** [comment]
Looks like `brew test-bot --only-cleanup-after` failed for the 10.14 run.

**Review by dawidd6 — APPROVED** [review]

**Pull request closed without merging**